server/scripts/import_roster.ts

```diff
@@ -0,0 +1,129 @@
+import axios from 'axios';
+import fsCoursesRead from 'fs';
+import csv from 'csv-parser';
+import { Course, RosterCourse, RosterResponse } from '../types';
+import { courses } from '../server';
+
+const currSem = 'FA21';
+const prevSem = 'SP21';
+const summerSem = 'SU21';
+
+const classRosterURL = `https://classes.cornell.edu/api/2.0/search/classes.json?roster=${currSem}&subject=`;
+
+const prevClassRosterURL = `https://classes.cornell.edu/api/2.0/search/classes.json?roster=${prevSem}&subject=`;
+
+const summerClassRosterURL = `https://classes.cornell.edu/api/2.0/search/classes.json?roster=${summerSem}&subject=`;
+
+const coursesCSV = [];
+
+fsCoursesRead
+  .createReadStream('./website_data_csv/courses.csv')
+  .pipe(csv())
+  .on('data', (data) => coursesCSV.push(data))
+  .on('end', () => {
+    const formattedCourses: string[][] = [];
+    for (let i = 0; i < coursesCSV.length; i += 1) {
+      const fCourse: string[] = coursesCSV[i].tag.split(' ');
+      formattedCourses.push(fCourse);
+    }
+    getRosterCourses(formattedCourses);
+  });
+
+async function getRosterCourses(targetCourses: string[][]) {
+  const fetchedCourses = [];
+  const missedCourses = [];
+  /* eslint-disable no-await-in-loop */
+  for (let i = 0; i < targetCourses.length; i += 1) {
+    await axios
+      .get(`${classRosterURL + courses[i][0]}&q=${courses[i][1]}`)
+      .then(async (res) => {
+        const resData: RosterResponse = (await res.data) as RosterResponse;
+        const classes = resData.data.classes as RosterCourse[];
+        fetchedCourses.push(classes[0] as RosterCourse);
+      })
+      .catch(() => {
+        missedCourses.push(courses[i]);
+      });
+  }
+  console.log('i finished getting courses from current roster');
```

Comment:
Probably remove `console.log` stuff before merging so that it's a bit more clear, unless we really need it